### PR TITLE
⬆️ Update example app to React Native 0.72.4

### PR DESCRIPTION
**Pull request by iujames**
Stashing some changes here for any feedback - testing out what it would take to include some [proposed Android dependency updates in the core SDK](https://github.com/appcues/appcues-android-sdk/pull/477), as part of the react native module.

Most notably, Kotlin 1.9 usage (Compose 1.5 requires) and compile SDK 34 (was 33) seem to require some update to the example app for react-native version > `0.69.9` that we had been using. It's not totally clear to me yet what the minimum version possible could be, but I went ahead and tried upgrading everything to the latest `0.72.4` to try it out, and it eventually worked.

A lot of configuration files needed to be changed, the[ upgrade helper](https://react-native-community.github.io/upgrade-helper/?from=0.69.9&to=0.72.4) assisted some, as did [creating a new native module project from scratch](https://reactnative.dev/docs/native-modules-setup) and copying some of the newer config files.

In short - these Android updates are feasible, but will likely require some of our customers to upgrade their React Native apps if they are not on a more recent version - so timing needs to be considered.

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -133,5 +133,5 @@ dependencies {
   implementation "org.jetbrains.kotlinx:kotlinx-coroutines-android:1.7.2"
 
   // Appcues
-  implementation "com.appcues:appcues:3.0.2"
+  implementation "com.appcues:appcues:3.0.3-beta01"
```
Comment (iujames) [inline android/build.gradle:136]: temporary - local package built from the Android branch with updates

**Comment by iujames** [comment]
@mmaatttt this should be good to go now for
```sh
$ npx react-native run-android
```
usage, using the latest local package from the android sdk after updates in https://github.com/appcues/appcues-android-sdk/pull/506.

The key update was the 3rd commit - reverting back to Kotlin 1.8.22 instead of 1.9.10, which will not work with react native at this point.

**Review by mmaatttt — APPROVED** [review]
